File: sdk/php/src/OmnidexApiException.php
<?php

declare(strict_types=1);

namespace Omnidex\Integration;

final class OmnidexApiException extends \RuntimeException
{
    public function __construct(
        public readonly int $status,
        public readonly string $apiMessage,
    ) {
        parent::__construct(sprintf(
            'Omnidex integration API failed with HTTP %d: %s',
            $status,
            $apiMessage,
        ));
    }
}

File: sdk/java/src/main/java/com/omnidex/integration/OmnidexApiException.java
package com.omnidex.integration;

public final class OmnidexApiException extends RuntimeException {
    private final int status;
    private final String apiMessage;

    public OmnidexApiException(int status, String apiMessage) {
        super("Omnidex integration API failed with HTTP " + status + ": " + apiMessage);
        this.status = status;
        this.apiMessage = apiMessage;
    }

    public int status() { return status; }
    public String apiMessage() { return apiMessage; }
}
